### PR TITLE
Draft contributing.md file

### LINKED ISSUE #4503: [Feature Request]: Create a CONTRIBUTING.md file for easier onboarding
### Platform

other

### Description

It would be nice to have a file like `CONTRIBUTING.md` that describes how to contribute to the project for faster and easier onboarding of new users. 

Example sections:
- How to build the project.
- What are the usual steps of adding a new board. 
- Any conventions used in the project that should be followed by everyone (add linters).

### PR DESCRIPTION
**Pull request by rcarteraz**
In response to a feature request I've created a draft CONTRIBUTING.md file to review. Welcome to feedback 

To close #4503

### CODE REVIEW

**Comment by fifieldt** [comment]
Probably worth informing people about the CLA and how to sign it?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+## Making Contributions
+
+> [!IMPORTANT]
+> Before making any contributions, you must sign our Contributor License Agreement (CLA). You can do this by visiting https://cla-assistant.io/meshtastic/firmware. Be sure to use the GitHub account you will use to submit your contributions when signing.
+
+1. Fork the repository
+2. Create a new branch for your feature or bug fix
+3. Make your changes
+4. Test your changes thoroughly
+5. Create a pull request with a clear description, using the provided template, of your changes
```
Comment (caveman99) [inline CONTRIBUTING.md:36]: Can we add a note about enabling 'maintainer edit access' here? It's really difficult to handle PR's that don't have this enabled.

grafik

grafik

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Coding Standards
+
+[Placeholder for coding standards]
```
Comment (caveman99) [inline CONTRIBUTING.md:40]: This is basically 'run trunk fmt before creating the PR' :-)

**Review by caveman99 — CHANGES_REQUESTED** [review]
Rest LGTM

**Review by caveman99 — APPROVED** [review]

**Pull request merged**